test/language/class/class-strict-mode.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class strict mode
+---*/
+class C {}
+
+with ({a: 1}) {
+  assert.sameValue(a, 1);
+}
+
+assert.throws(SyntaxError,
+             'class C extends function B() { with ({}); return B; }() {}');
```

Comment:
It doesn't look like Test262's implementation of `assert.throws` accepts a string: https://github.com/tc39/test262/blob/ba3be4a575b453bd9792ecbfc5e4e844b2422434/harness/assert.js#L44-L64